### PR TITLE
[v7.5.x] Settings: Fix handling

### PR DESCRIPTION
**Pull request by joanlopez**
**Special notes for your reviewer**:

Manual backport of https://github.com/grafana/grafana/pull/42497#issuecomment-982501981.

I did some tests and it seems to be working, but feel free to test it yourself as well.

### CODE REVIEW

File: pkg/setting/setting_test.go [hunk 1]
```diff
@@ -78,16 +78,6 @@ func TestLoadingSettings(t *testing.T) {
...
-		Convey("Should return an error when url is invalid", func() {
-			err := os.Setenv("GF_DATABASE_URL", "postgres.%31://grafana:secret@postgres:5432/grafana")
-			require.NoError(t, err)
-
-			cfg := NewCfg()
-			err = cfg.Load(&CommandLineArgs{HomePath: "../../"})
-
-			So(err, ShouldNotBeNil)
-		})
-
```
Comment (joanlopez) [inline pkg/setting/setting_test.go:90]: As part of security fix we are redacting all URLs and not only database one, so it no longer makes sense to do this kind of handling specifically for database URL, I think.

File: pkg/setting/setting_test.go [hunk 2]
```diff
@@ -96,7 +86,7 @@ func TestLoadingSettings(t *testing.T) {
 			err = cfg.Load(&CommandLineArgs{HomePath: "../../"})
 			So(err, ShouldBeNil)
 
-			So(appliedEnvOverrides, ShouldContain, "GF_DATABASE_URL=mysql://user:-redacted-@localhost:3306/database")
+			So(appliedEnvOverrides, ShouldContain, "GF_DATABASE_URL=mysql://user:xxxxx@localhost:3306/database")
```
Comment (joanlopez) [inline pkg/setting/setting_test.go:89]: For the sake of security (and some other reasons), now we rely on Go standard library function to redact URLs, which uses `xxxxx` as placeholder, which I guess it's fine.

**Review by undef1nd — APPROVED** [review]
I've tested on a few settings and seems to work fine.

**Review by xlson — CHANGES_REQUESTED** [review]
I looked over my outstanding PR reviews and found this oldie. Seems to me that it still would be a good idea to try to get this merged.

**Review by xlson — APPROVED** [review]

**Pull request merged**